warehouse/accounts/services.py

```diff
@@ -63,6 +74,42 @@ def get_user(self, userid):
         # TODO: We need some sort of Anonymous User.
         return self.db.query(User).get(userid)
 
+    def get_user_by_otk(self, otk):
+        if not otk:
+            raise InvalidPasswordResetToken
+
+        try:
+            data = self.serializer.loads(
+                otk,
+                self.token_max_age
+            )
+        except BadData:
```

Comment:
Regarding expiry, I would wonder if we can determine failure reason here. The current expiration TTL of 6 hours will probably lead to a few users wondering WTF happened when they submit an expired token.

Reply:
+1 on this- @di if you are happy to output a message on the page, I can style it.